packages/jest-haste-map/src/index.ts

```diff
@@ -846,6 +846,17 @@ class HasteMap extends EventEmitter {
         return;
       }
 
+      const relativeFilePath = fastPath.relative(rootDir, filePath);
+      const fileMetadata = hasteMap.files.get(relativeFilePath);
+
+      // The file has been accessed, not modified
+      if (
+        type === 'change' &&
+        fileMetadata?.[H.MTIME] === stat?.mtime.getTime()
```

Comment:
what if both are `undefined`? do we still wanna return here?

Reply:
Um, that's a valid point. It used to check for `fileMetadata ` and `stat ` separately, but I thought it's going to be clearer with optional chaining. Will update